src/BayesFilters/include/BayesFilters/utils.h

```diff
@@ -83,6 +83,102 @@ Eigen::VectorXd multivariate_gaussian_log_density(const Eigen::MatrixBase<Derive
 }
 
 
+/**
+ * Evaluate the logarithm of a multivariate Gaussian probability density function
+ * with high-dimensional input using the Sherman–Morrison–Woodbury formula.
+ *
+ * It is assumed that the covariance matrix S can be written in the form S = UV + R
+ * where R is an invertible block diagonal matrix. Each block R_{i} is square and has
+ * dimension M such that N * M is the size of the input for some positive integer N.
+ *
+ * This version is much faster than the standard gaussian logarithm density evaluation
+ * if U.cols() << U.rows().
+ *
+ * @param input Input representing the argument of the function as a vector or matrix.
+ * @param mean The mean of the associated Gaussian distribution as a vector.
+ * @param U The U factor within the covariance matrix S = UV + R of the associated Gaussian distribution as a matrix.
+ * @param V The V factor within the covariance matrix S = UV + R of the associated Gaussian distribution as a matrix.
+ * @param R The R summand within the covariance matrix S = UV + R of the associated Gaussian distribuion as a matrix.
+ *        The matrix R has size M * (N * M) and consists in the concatenation of all the diagonal blocks R_{i}.
+ *        If the diagonal blocks are all equal, it is possible to provide a single M * M block matrix.
+ *
+ * @return The value of the logarithm of the density function evaluated on the input data as a vector.
+ */
+template<typename Derived>
+Eigen::VectorXd multivariate_gaussian_log_density_high_dim(const Eigen::MatrixBase<Derived>& input, const Eigen::Ref<const Eigen::VectorXd>& mean, const Eigen::Ref<const Eigen::MatrixXd>& U, const Eigen::Ref<const Eigen::MatrixXd>& V, const Eigen::Ref<const Eigen::MatrixXd>& R)
```

Comment:
I don't quite get the name of this function. Wouldn't it be more informative if it would be something like `multivariate_gaussian_log_density_UVR`? This function does not account, in general, high dimensional matrices, but ones that can be decomposed in a particular form. I would write that "form factor" information in the function name 😄

Reply:
I agree, being high dimensional is just a particular case. Do you want me to change the title of the PR and the associated issue?

Reply:
Yes please! 👍